flink-runtime/src/main/java/org/apache/flink/runtime/state/v2/StateDescriptor.java

```diff
@@ -0,0 +1,167 @@
+ * State in stateful operations internally.
+ *
+ * @param <T> The type of the value of the state object described by this state descriptor.
+ */
+@Internal
+public abstract class StateDescriptor<T> implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /** An enumeration of the types of supported states. */
+    public enum Type {
+        VALUE,
+        LIST,
+        REDUCING,
+        FOLDING,
+        AGGREGATING,
+        MAP
+    }
+
+    /** ID that uniquely identifies state created from this StateDescriptor. */
+    @Nonnull private final String stateId;
+
+    /** The serializer for the type. */
+    @Nonnull private final TypeSerializer<T> typeSerializer;
+
+    /**
+     * The type information describing the value type. Remain this since it could provide more
+     * information which could be used internally in the future.
+     */
+    @Nonnull private final TypeInformation<T> typeInfo;
+
+    /** The configuration of state time-to-live(TTL), it is disabled by default. */
+    @Nonnull private StateTtlConfig ttlConfig = StateTtlConfig.DISABLED;
+
+    // ------------------------------------------------------------------------
+
+    /**
+     * Create a new {@code StateDescriptor} with the given stateId and the given type information.
+     *
+     * @param stateId The stateId of the {@code StateDescriptor}.
+     * @param typeInfo The type information for the values in the state.
+     */
+    protected StateDescriptor(@Nonnull String stateId, TypeInformation<T> typeInfo) {
+        this(stateId, typeInfo, new SerializerConfigImpl());
+    }
+
+    /**
+     * Create a new {@code StateDescriptor} with the given stateId and the given type information.
+     *
+     * @param stateId The stateId of the {@code StateDescriptor}.
+     * @param typeInfo The type information for the values in the state.
+     * @param serializerConfig The serializer related config used to generate {@code
+     *     TypeSerializer}.
+     */
+    protected StateDescriptor(
+            @Nonnull String stateId,
+            @Nonnull TypeInformation<T> typeInfo,
+            SerializerConfig serializerConfig) {
+        this.stateId = checkNotNull(stateId, "stateId must not be null");
+        this.typeInfo = checkNotNull(typeInfo, "type information must not be null");
+        this.typeSerializer = typeInfo.createSerializer(serializerConfig);
+    }
+
+    // ------------------------------------------------------------------------
+
+    /**
+     * Configures optional activation of state time-to-live (TTL).
+     *
+     * <p>State user value will expire, become unavailable and be cleaned up in storage depending on
+     * configured {@link StateTtlConfig}.
+     *
+     * @param ttlConfig configuration of state TTL
+     */
+    public void enableTimeToLive(StateTtlConfig ttlConfig) {
+        this.ttlConfig = Preconditions.checkNotNull(ttlConfig);
```

Comment:
Since you have `import static Preconditions.checkNotNull`, how about remove the `Preconditions`?